src/java/org/apache/cassandra/config/GuardrailsOptions.java

```diff
@@ -474,6 +475,30 @@ public void setItemsPerCollectionThreshold(int warn, int fail)
                                   x -> config.items_per_collection_fail_threshold = x);
     }
 
+    public int getFieldsPerUDTWarnThreshold()
```

Comment:
I think those writes into volatile fields are safe since they are inside a constructor called on initialization, so there shouldn't be any other callers.